File: c/homework/stack.c
#include <stdio.h>

int stack[2];
int index = -1;
int length = sizeof(stack) / sizeof(stack[0]);  


int isFull() {
    if(index >= length - 1)
        return 1;
    else
        return 0;
}


int isEmpty() {
    if(index < 0)
        return 1;
    else
        return 0;
}

void push(int n) {
    if(!isFull()) {
        index++;
        stack[index] = n;
    } else {
        printf("Stack overflow\n");
    }
}

int pop() {
    if(!isEmpty()) {
        int value = stack[index];
        index--;
        return value;
    } else {
        printf("Stack is empty\n");
        return -1;
    }
}

void displayStack() {
    if(!isEmpty()) {
        for (int i = 0; i <= index; i++) {
            printf("%d ", stack[i]);
        }
        printf("\n"); 
    } else {
        printf("Stack is empty\n");
    }
}

int main() {
 
    push(11);
    push(13);
    push(156);

    
    printf("Stack after pushes: ");
    displayStack();

  
    printf("Popped value: %d\n", pop());
    printf("Stack after pop: ");
    displayStack();
    
    return 0;
}

File: js/stack.js
const stack = [];
let index = -1;
const length = 10;

const isFull = () => {
  return index >= length - 1;
};

const isEmpty = () => {
  return index < 0;
};

const push = (n) => {
  if (!isFull()) {
    index++;
    stack[index] = n;
  } else {
    console.log("Stack is full");
  }
};

const pop = () => {
  if (!isEmpty()) {
    const value = stack[index];
    index--;
    return value;
  } else {
    console.log("Stack is empty");
    return -1;
  }
};

const displayStack = () => {
  if (isEmpty()) {
    console.log("Stack is empty");
    return;
  }
  for (let i = 0; i <= index; i++) {
    process.stdout.write(stack[i] + " ");
  }
  console.log();
};

push(10);
push(20);
push(30);

console.log("Stack after pushes: ");
displayStack();

console.log("Popped values: ", pop(), pop());
console.log("Stack after pops: ");
displayStack();
